lib/include/openturns/OTtestcode.hxx

```diff
@@ -1067,6 +1066,14 @@ private:
   UnsignedInteger fittingSamplingSize_ = 100;
 };
 
+void assert_condition(const bool & condition, const String errMsg = "")
+{
+  if (!condition)
+  {
+    throw TestFailed(OSS() << "Value " << condition << " is not true :" << " " << errMsg);
+  }
+}
+
```

Comment:
the python build-on assert method already does exactly that
and in c++ you can already use assert_equal with boolean arguments
so please remove this

Reply:
This is not in Python, it is in C++. The reason I created this function is because the assert function does not perform its job. In the C++ unit test `lib/test/t_FunctionalChaos_conditionalExpectation.cxx`, I wrote:
```
assert(false);
```
and the test still passes, which is obviously wrong. I noticed this because there was a temporary error in the C++ test, that did not generate a failing test. Please test this and you will see by yourself. Do you think that this is related to the build options I used?

Reply:
yes, the C assert does not do anything depending on the build type, dont use it
my previous comment still stands

Reply:
Ok, fixed at 866fae63cea27aa92bce6cb85725119c6351803c

However, the `assert` function is used in other unit tests: I will fix this in a separate PR.